sklearn/linear_model/_coordinate_descent.py

```diff
@@ -146,47 +149,35 @@ def _alpha_grid(
             "your estimator with the appropriate `alphas=` "
             "argument."
         )
-    n_samples = len(y)
-
-    sparse_center = False
-    if Xy is None:
-        X_sparse = sparse.isspmatrix(X)
-        sparse_center = X_sparse and (fit_intercept or normalize)
-        X = check_array(
-            X, accept_sparse="csc", copy=(copy_X and fit_intercept and not X_sparse)
```

Comment:
I think this is valuable input checking that we should keep.

Reply:
(1) `_alpha_grid` is only called from `LinearModelCV` (which always checks `X` and `y`) or `enet_path`, which also checks them if `check_input is True`.
(2) For this reason, I'm calling `_preprocess_data` with `check_input=False`. Even if you wanted to repeat the checking, it seems like we would just let `_preprocess_data` handle the checking.

Reply:
ok